util/errors/errors.go

```diff
@@ -0,0 +1,55 @@
+package errors
+
+import (
+	"net"
+	"net/url"
+	"strings"
+
+	apierr "k8s.io/apimachinery/pkg/api/errors"
+
+	argoerrs "github.com/argoproj/argo/errors"
+)
+
+func IsTransientErr(err error) bool {
```

Comment:
This looks very similar to `IsRetryableKubeAPIError` and the new name tell you there is a change to its semantics:

- Removed `IsNotFound`, `IsInvalid`, `IsMethodNotSupported`  conditions, because I don't think these are  transient. Am I right or wrong?
- Removed `IsForbidden` condition, we only check for `exceeded quota` now. 
- Added `isTransientNetworkErr`to catch network errors (this is copy-and-paste of `IsRetryableNetworkError `)